### PR TITLE
Allow for filtering of CSS Styles

### PR DESCRIPTION
**Pull request by cwill747**
Adds the ability to parse _inline_ css and decide whether or not to include it based off of the tag names. Uses the css library to parse out the inline css, filters it, then stringify's the CSS and re-inserts it, meaning that some whitespace or semicolons could be lost on the reparsing.

Definitely needs some code review.

### CODE REVIEW

**Comment by boutell** [comment]
Haven't had time to look at this but certainly very interested considering the XSS risks of both inline CSS and the style tag.

**Comment by niftylettuce** [comment]
+1

**Comment by pschwyter** [comment]
+1

**Comment by mloschia** [comment]
Can someone merge this please?

**Comment by cwill747** [comment]
Conflicts resolved, but the `css` npm project I depended on when I wrote this seems to be long-dead, and people are using postcss now. I'll refactor to use, since postcss is kept way more up to date

File: README.md [hunk 1]
```diff
@@ -310,6 +310,29 @@ clean = sanitizeHtml(dirty, {
...
+
+```javascript
+clean = sanitizeHtml(dirty, {
+        allowedTags: ['p'],
+        allowedAttributes: {
+          'p': ["style"],
+        },
+        allowedStyles: {
+          '*': {
+            'color': '*',
```
Comment (pschwyter) [inline README.md]: can we allow users to specify a regexp? A wildcard defeats the point of having sanitization

Comment (pschwyter) [reply README.md]: probably *do not* want to  allow a wildcard

File: index.js [hunk 2]
```diff
@@ -373,6 +388,41 @@ function sanitizeHtml(html, options, _recursing) {
...
+        });
+        originalAst.stylesheet.rules[0].declarations = allowedDeclarations;
+        return originalAst;
+      } else {
+        return false;
+      }
+    } else {
+      return originalAst;
+    }
+  }
```
Comment (mloschia) [inline index.js]: This method could be broken out into several smaller functions to make it more dry. I don't mind volunteering to do it.

File: README.md (hunk 1 above)
Comment (mloschia) [reply README.md]: @pschwyter agreed, a wild card would introduce security concerns. That would enable anyone to enter malicious code

Comment (mloschia) [reply README.md]: RegExps would also enable whitelisting of measurement values. For example, if we wanted to whitelist pixels within font-size, all we'd need to do is append that to the regex.

File: index.js (hunk 2 above)
Comment (mloschia) [reply index.js]: Also this is where we should do validation on the values of these css properties. It's be easy to do if Regexps are part of our whitelist

Comment (cwill747) [reply index.js]: I'll refactor on postcss and leave the function as-is, and then if you wanna chop it up, go for it!

Comment (mloschia) [reply index.js]: @cwill747 sounds good!

**Comment by mloschia** [comment]
@cwill747 I did the package update and refactor. https://github.com/punkave/sanitize-html/pull/178

**Pull request merged**